src/metax_api/api/oaipmh/base/metax_oai_server.py

```diff
@@ -50,10 +50,12 @@ def _get_oai_dc_urnresolver_metadata(self, record):
         Preferred identifier is added only for ida and att catalog records
         other identifiers are added for all.
         """
+        preferred_identifier = record.research_dataset.get('preferred_identifier')
```

Comment:
Does not warrant a change, but for future convenience: preferred_identifier can be accessed like record.preferred_identifier in a similar way like you used urn_identifier before.